src/create-with-theme.js

```diff
@@ -0,0 +1,39 @@
+const React = require('react');
+const PropTypes = require('prop-types');
+const channel = require('./channel');
+
+function createWithTheme(CHANNEL = channel) {
+  return Component =>
+    class withTheme extends React.Component {
+      static contextTypes = {
+        [CHANNEL]: PropTypes.func,
+      };
+
+      componentWillMount() {
+        if (!this.context[CHANNEL]) {
+          throw new Error(
```

Comment:
That wouldn't show anything useful though because components underneath would do `props => props.theme.colors.primary` and then everything crashes anyway and you'd get an error `"Cannot read property colors of undefined"`. I'd much rather see `"You forgot to use a ThemeProvider"`.